apache/metron
# METRON-1695: Expose pcap properties through Ambari

## Contributor Comments
This change makes the PCAP config properties available via the Metron service -> Configs tab in Ambari. It also takes care of updating the `pcap.properties` file with relevant values depending on the deploy environment (single node vs. multi node vs. kerberized cluster). 

Here's a screenshot of the PCAP Config:
![pcap-config-values](https://user-images.githubusercontent.com/20395490/45877981-21e62680-bdbd-11e8-902d-6d5a2a1bc239.png)

### Testing Steps
* Launch full dev
* Metron PCAP service should come up successfully.
* Go to Ambari -> Metron services -> Configs -> PCAP tab; change any config parameter; 
* Restart the required services and the config change take effect.

### Testing Done
I ran the following different tests to validate the fix.

#### A. Full Dev
* Verified Metron PCAP service comes up successfully 
* Changed some of the PCAP config values from the Ambari UI, validated that this is reflected in the `pcap.properties` file upon service restart.
- **Note**: I ran into [METRON-1786](https://issues.apache.org/jira/browse/METRON-1786) during testing. I was able to resolve the issue by applying the fix in PR #1206 

#### B. Multi-node Cluster
* Validated that the `kafka.zk` property in `pcap.properties` is updated with the correct Broker details
* Changed a PCAP service-related property (E.g. `kafka_pcap_numpackets`) and validated that the `pcap.properties` file is updated.
* Changed a PCAP REST-related property (E.g. `pcap_page_size`) and validated that the `/etc/default/metron` file is updated.

#### C. Kerberized Cluster
* Kerberized the multi-node cluster and setup pycapa
* Validated that `topology.worker.childopts` and `topology.auto-credentials` are 
* Ingested PCAP into the cluster
* Ran queries through the PCAP panel and validated that the modified config settings are in effect (E.g. `pcap_page_size` set to 50). 

## Pull Request Checklist

Thank you for submitting a contribution to Apache Metron.  
Please refer to our [Development Guidelines](https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=61332235) for the complete guide to follow for contributions.  
Please refer also to our [Build Verification Guidelines](https://cwiki.apache.org/confluence/display/METRON/Verifying+Builds?show-miniview) for complete smoke testing guides.  


In order to streamline the review of the contribution we ask you follow these guidelines and ask you to double check the following:

### For all changes:


### For code changes:
- [NA] Have you ensured that the full suite of tests and checks have been executed in the root metron folder via:
  ```
  mvn -q clean integration-test install && dev-utilities/build-utils/verify_licenses.sh 
  ```


### For documentation related changes:

  ```
  cd site-book
  mvn site
  ```

#### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.
It is also recommended that [travis-ci](https://travis-ci.org) is set up for your personal repository such that your branches are built there before submitting a pull request.
